examples/plot_log_scale/src/main.rs

```diff
@@ -0,0 +1,154 @@
+            },
+            Signal {
+                name: "y=x^2",
+                points: x.clone().map(|x| [x, x.powi(2)]).collect(),
+            },
+            Signal {
+                name: "y=exp(x)",
+                points: x.clone().map(|x| [x, x.exp()]).collect(),
+            },
+            Signal {
+                name: "y=ln(x)",
+                points: x.clone().map(|x| [x, x.ln()]).collect(),
+            },
+        ];
+        Self {
+            log_x: true,
+            log_y: true,
+            signals,
+        }
+    }
+}
+
+impl eframe::App for PlotExample {
+    fn update(&mut self, ctx: &egui::Context, _: &mut eframe::Frame) {
+        egui::SidePanel::left("options").show(ctx, |ui| {
+            ui.checkbox(&mut self.log_x, "X axis log scale");
+            ui.checkbox(&mut self.log_y, "Y axis log scale");
+        });
+        let log_x = self.log_x;
+        let log_y = self.log_y;
+        egui::CentralPanel::default().show(ctx, |ui| {
+            let mut plot = egui_plot::Plot::new("plot")
+                .legend(Legend::default())
+                .label_formatter(|name, value| {
+                    let x = if log_x {
+                        10.0f64.powf(value.x)
+                    } else {
+                        value.x
+                    };
+                    let y = if log_y {
+                        10.0f64.powf(value.y)
+                    } else {
+                        value.y
+                    };
+                    if !name.is_empty() {
+                        format!("{name}: {x:.3}, {y:.3}")
+                    } else {
+                        format!("{x:.3}, {y:.3}")
+                    }
+                });
+            if log_x {
+                plot = plot
+                    .x_grid_spacer(log_axis_spacer)
+                    .x_axis_formatter(log_axis_formatter);
+            }
+            if log_y {
+                plot = plot
+                    .y_grid_spacer(log_axis_spacer)
+                    .y_axis_formatter(log_axis_formatter);
+            }
+            plot.show(ui, |plot_ui| {
+                for signal in &self.signals {
+                    let points: Vec<_> = signal
+                        .points
+                        .iter()
+                        .copied()
+                        .map(|[x, y]| {
+                            let x = if log_x { x.log10() } else { x };
+                            let y = if log_y { y.log10() } else { y };
+                            [x, y]
+                        })
+                        .collect();
+                    plot_ui.line(Line::new(points).name(signal.name));
+                }
+            });
+        });
+    }
+}
+
+#[allow(clippy::needless_pass_by_value)]
+fn log_axis_spacer(input: GridInput) -> Vec<GridMark> {
+    let (min, max) = input.bounds;
+    let mut marks = vec![];
+    for i in min.floor() as i32..=max.ceil() as i32 {
```

Comment:
I think the only thing missing is to make this spacer more general. Right now it only works well if the number of decades visible on the axis is between 0.5 and 15, roughly. It is a decent range for a log plot, but it would be nice if it could have the "infinite" zoom-in and zoom-out like the default grid. I haven't been able to figure that out yet.

Reply:
I had implemented it on top of the original plot, here's what I came up with

```
pub fn log_grid_spacer(log_base: i64) -> GridSpacer {
    let log_base = log_base as f64;
    let step_sizes = move |input: GridInput| -> Vec<GridMark> {
        // The distance between two of the thinnest grid lines is "rounded" up
        // to the next-bigger power of base
        if input.log {
            let smallest_power = last_power(input.bounds.0, log_base)
                .log(log_base)
                .ceil()
                .max(0.0) as i32;
            let largest_power = next_power(input.bounds.1, log_base).log(log_base).ceil() as i32;

            let mut steps = vec![];
            for power in smallest_power..largest_power {
                let step_size = log_base.powi(power);
                let next_step_size = log_base.powi(power + 1);

                let bounds = (
                    step_size.max(input.bounds.0),
                    next_step_size.min(input.bounds.1),
                );

                if bounds.0 >= bounds.1 {
                    dbg!("oopsie");
                }

                fill_marks_between(&mut steps, step_size, bounds);
            }
            steps
        } else {
            let smallest_visible_unit = next_power(input.base_step_size, log_base);

            let step_sizes = [
                smallest_visible_unit,
                smallest_visible_unit * log_base,
                smallest_visible_unit * log_base * log_base,
            ];

            generate_marks(step_sizes, input.bounds)
        }
    };

    Box::new(step_sizes)
}

fn next_power(value: f64, base: f64) -> f64 {
    assert_ne!(value, 0.0); // can be negative (typical for Y axis)
    base.powi(value.abs().log(base).ceil() as i32)
}

fn last_power(value: f64, base: f64) -> f64 {
    assert_ne!(value, 0.0); // can be negative (typical for Y axis)
    base.powi(value.abs().log(base).floor() as i32)
}
```

Reply:
Thanks, I'll see if I can adapt this to what I was trying to do (since some of these egui_plot functions are private). I'm still not sure if this is better off as an example or as part of the API. Either way feels more complicated than it should be with all the options people might need.